lua.go

```diff
@@ -0,0 +1,178 @@
+package crossplane
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Lua adds support for directives added to NGINX by the ngx_http_lua_module module.
+//
+// Lua implements the Lexer interface by tokenizing *_by_lua_block directives with a
+// simple Lua parser. The Lua blocks will be placed into a token to be set as an
+// argument on the Lua directive.
+//
+// Lua also implements the Builder interface by writing the *_by_lua_block
+// directive's contents into the directive's block.
+type Lua struct{}
+
+// directiveNames returns a list of Lua module directive names used in NGINX configurations.
+func (l *Lua) directiveNames() []string {
+	return []string{
+		"init_by_lua_block",
+		"init_worker_by_lua_block",
+		"exit_worker_by_lua_block",
+		"set_by_lua_block",
+		"content_by_lua_block",
+		"server_rewrite_by_lua_block",
+		"rewrite_by_lua_block",
+		"access_by_lua_block",
+		"header_filter_by_lua_block",
+		"body_filter_by_lua_block",
+		"log_by_lua_block",
+		"balancer_by_lua_block",
+		"ssl_client_hello_by_lua_block",
+		"ssl_certificate_by_lua_block",
+		"ssl_session_fetch_by_lua_block",
+		"ssl_session_store_by_lua_block",
+	}
+}
+
+// RegisterLexer registers a lexer for parsing Lua blocks.
+func (l *Lua) RegisterLexer() RegisterLexer { //nolint:ireturn
+	return LexWithLexer(l, l.directiveNames()...)
+}
+
+// Lex lexically analyzes the Lua blocks based on directives detected.
+// It is used by the lexer to tokenize Lua content within configuration files.
+//
+//nolint:funlen,gocognit,gocyclo,nosec
```

Comment:
nit, it's totally new code, I'm wondering if we could do better?

Reply:
We are following the Python version of Crossplane https://github.com/nginxinc/crossplane/blob/master/crossplane/ext/lua.py  to implement the Lua part. While there might be some optimizations for this part, considering time constraints, perhaps we could have another PR dedicated to refactoring and optimization.